### ISSUE TITLE
ostree commit ID

### ISSUE THREAD

**Issue by vishnukumarts** [ostreedev/ostree#893]
Why the commit id is getting different if i have not mentioned --skip-if-unchanged parameter.  Even the same source is committed in two different PC, ID is getting different.

**Comment by cgwalters** [comment]
I suspect it's something like Python `.py` files with embedded timestamps.  Another common cause is (for the ostree-for-host case) the initramfs (at least with dracut).

You can use `ostree diff` to find the difference between the two commits - looking at the content changes in those files should help you debug.

**Comment by cgwalters** [comment]
See for example [this code in flatpak](https://github.com/flatpak/flatpak/blob/e69237f79208e9d83a6453ed2061fedb56ce88b5/builder/builder-post-process.c#L116) which post-processes `.pyc` files.

For [rpm-ostree](https://github.com/projectatomic/rpm-ostree) because we're just committing pregenerated RPMs, we don't have the `.pyc` problem, but we do have the initramfs issue, plus there are some packages which drop certificates in `/etc` which have timestamps.

Basically, making `ostree commit --skip-if-unchanged` work is going to require lots of timestamp issue fixing in your OS content and/or your build process.
